backend/compact-connect/lambdas/purchases/data_model/client.py

```diff
@@ -19,3 +24,99 @@ def get_privilege_purchase_options(self, *, compact: str, dynamo_pagination: dic
             KeyConditionExpression=Key('pk').eq(f'{compact}#CONFIGURATION'),
             **dynamo_pagination,
         )
+
+    @paginated_query
+    def get_provider(
+        self,
+        *,
+        compact: str,
+        provider_id: str,
+        dynamo_pagination: dict,
+        detail: bool = True,
+        consistent_read: bool = False,
+    ):
+        logger.info('Getting provider', provider_id=provider_id)
+        if detail:
+            sk_condition = Key('sk').begins_with(f'{compact}#PROVIDER')
+        else:
+            sk_condition = Key('sk').eq(f'{compact}#PROVIDER')
+
+        resp = self.config.provider_table.query(
+            Select='ALL_ATTRIBUTES',
+            KeyConditionExpression=Key('pk').eq(f'{compact}#PROVIDER#{provider_id}') & sk_condition,
+            ConsistentRead=consistent_read,
+            **dynamo_pagination,
+        )
+        if not resp['Items']:
+            raise CCNotFoundException('Provider not found')
+
+        return resp
+
+    def _generate_privilege_record(
+        self,
+        compact_name: str,
+        provider_id: str,
+        jurisdiction_postal_code: str,
+        license_expiration_date: date,
+        compact_transaction_id: str,
+    ):
+        privilege_object = {
+            'providerId': provider_id,
+            'compact': compact_name,
+            'jurisdiction': jurisdiction_postal_code.lower(),
+            'status': 'active',
+            'dateOfIssuance': datetime.now(tz=UTC).date(),
+            'dateOfExpiration': license_expiration_date,
+            'compactTransactionId': compact_transaction_id,
+        }
+        schema = PrivilegeRecordSchema()
+        return schema.dump(privilege_object)
+
+    def create_provider_privileges(
```

Comment:
Should we have some unit/function tests targeting this method in particular?

Reply:
but I will add some unit tests testing the error handling flow